src/components/Wizard/CreateVmWizard/tests/NetworksTab.test.js

```diff
@@ -73,4 +103,99 @@ describe('<NetworksTab />', () => {
     expect(isBootableNetwork(networks[1])).toBeFalsy();
     expect(isBootableNetwork(networks[2])).toBeTruthy();
   });
+
+  it('requires bootable network for PXE image source', () => {
+    const component = mount(testNetworksTab());
+    expect(getTableRows(component)).toHaveLength(0);
+
+    getNetworkButton(component).simulate('click');
+    const rows = getTableRows(component);
+    expect(rows).toHaveLength(1);
+    const columns = rows.find('td');
+
+    expect(
+      columns
+        .at(0)
+        .find('input')
+        .props().defaultValue
+    ).toEqual('eth0');
+
+    expect(
+      columns
+        .at(1)
+        .find('input')
+        .props().defaultValue
+    ).toBeUndefined();
+    expect(
+      columns
+        .at(1)
+        .find('input')
+        .props().value
+    ).toBeUndefined();
+
+    const dropdownItems = columns
+      .at(2)
+      .find(Dropdown)
+      .find(MenuItem);
+    expect(dropdownItems).toHaveLength(2);
+    expect(dropdownItems.at(0).text()).toEqual(NetworksTabFixture.props.networkConfigs[0].metadata.name);
+    expect(dropdownItems.at(1).text()).toEqual(POD_NETWORK);
+
+    expect(
+      columns
+        .at(2)
+        .find(Dropdown)
+        .props().value
+    ).toEqual(SELECT_NETWORK);
+
+    expect(
+      component
+        .find('#pxe-nic-dropdown')
+        .find(Dropdown)
+        .props().value
+    ).toEqual(SELECT_PXE_NIC);
+    expect(component.find(HelpBlock).text()).toEqual(PXE_ERROR);
+
+    // add PXE-bootable network
+    component.instance().rowsChanged([pxeRow], false);
+    component.update();
+
+    expect(
+      component
+        .find('#pxe-nic-dropdown')
+        .find(Dropdown)
+        .props().value
+    ).toEqual('pxe-net');
+    expect(component.find(HelpBlock).text()).toEqual('');
+
+    // add Pod network - not PXE bootable
+    component.instance().rowsChanged([podRow], false);
+    component.update();
+
+    expect(
+      component
+        .find('#pxe-nic-dropdown')
+        .find(Dropdown)
+        .props().value
+    ).toEqual(SELECT_PXE_NIC);
+    expect(component.find(HelpBlock).text()).toEqual(PXE_ERROR);
+
+    // add Pod network and PXE-bootable network
+    component.instance().rowsChanged([podRow, pxeRow], false);
+    component.update();
```

Comment:
just a small detail: we could also check onChange callbacks if they return validation

Reply:
added checks for onChange callbacks